src/main/java/io/fabric8/jenkins/openshiftsync/BuildConfigWatcher.java

```diff
@@ -96,6 +97,18 @@
   private Watch buildConfigWatch;
   private ScheduledFuture relister;
 
+  private static final ConcurrentHashSet<String> deletesInProgress = new ConcurrentHashSet<String>();
+
+  public static void deleteInProgress(String bcName) {
+    deletesInProgress.add(bcName);
+  }
+  public static boolean isDeleteInProgress(String bcID) {
+    return deletesInProgress.contains(bcID);
+  }
+  public static void deleteCompleted(String bcID) {
```

Comment:
same thing here ... if you want to fully implement, I would by a try finally in around the job.delete() call in `deleteJob`, where you call `deleteInProgress` right before `job.delete()`, and you call `deleteCompleted` in the finally block.